pynumaflow/function/server.py

```diff
@@ -0,0 +1,123 @@
+import asyncio
+import logging
+from os import environ
+
+from pynumaflow.function.generated import udfunction_pb2
+from pynumaflow.function.generated import udfunction_pb2_grpc
+
+from google.protobuf import empty_pb2 as _empty_pb2
+
+import grpc
+from typing import Callable, Any, Iterator
+
+from pynumaflow._constants import (
+    FUNCTION_SOCK_PATH,
+    DATUM_KEY,
+)
+from pynumaflow.function import Messages
+from pynumaflow.datum import Datum
+from pynumaflow.types import NumaflowServicerContext
+
+if environ.get("PYTHONDEBUG"):
+    logging.basicConfig(level=logging.DEBUG)
+
+_LOGGER = logging.getLogger(__name__)
+
+UDFMapCallable = Callable[[str, Datum, Any], Messages]
+
+
+class UserDefinedFunctionServicer(udfunction_pb2_grpc.UserDefinedFunctionServicer):
+    """
+    Provides an interface to write a User Defined Function (UDFunction)
+    which will be exposed over gRPC.
+
+    Args:
+        map_handler: Function callable following the type signature of UDFMapCallable
+        sock_path: Path to the UNIX Domain Socket
+
+    Example invocation:
+    >>> from pynumaflow.datum import Datum
+    >>> from pynumaflow.function import Messages, Message
+    >>> from pynumaflow.function.server import UserDefinedFunctionServicer
+    >>> def map_handler(key: str, datum: Datum) -> Messages:
+    ...   val = datum.value
+    ...   _ = datum.event_time
+    ...   _ = datum.watermark
+    ...   messages = Messages()
+    ...   messages.append(Message.to_vtx(key, val))
+    ...   return messages
+    >>> grpc_server = UserDefinedFunctionServicer(map_handler)
+    >>> grpc_server.start()
+    """
+
+    def __init__(self, map_handler: UDFMapCallable, sock_path=FUNCTION_SOCK_PATH):
+        self.__map_handler: UDFMapCallable = map_handler
+        self.sock_path = sock_path
+        self._cleanup_coroutines = []
+
+    def MapFn(
+        self, request: udfunction_pb2.Datum, context: NumaflowServicerContext
+    ) -> udfunction_pb2.DatumList:
+        """Applies a function to each datum element."""
```

Comment:
Can we add a comment inside the docstring that it is named like this to be compliant with the protobuf library? That's why we are keeping it `MapFn` right?